### PR TITLE
Retry on 412 status codes

### LINKED ISSUE #332: [BUG] Retry in case of 412 errors
**Describe the bug**

With Vault 1.10, a 412 status code is returned from Vault if you issue a request to a performance standby node right after obtaining a token via login. The action fails when it sees the 412 status code instead of retrying the request.

See [new-error-response-for-login-requests-to-perf-standbys-lagging-behind-active-node](https://www.vaultproject.io/docs/upgrading/upgrade-to-1.10.x#new-error-response-for-login-requests-to-perf-standbys-lagging-behind-active-node)

**To Reproduce**

Use the vault-action step with jwt authentication against Vault. You must have Vault running with performance standby nodes.

**Expected behavior**

The action should retry the requests that return a 412 status code. See https://www.vaultproject.io/docs/faq/ssct#q-is-there-anything-else-i-need-to-consider-to-achieve-consistency-besides-upgrading-to-vault-1-10

**Log Output**

```
2022-06-13T08:11:34.8806297Z ::group::Get Vault Secrets
2022-06-13T08:11:34.8806631Z ##[group]Get Vault Secrets
2022-06-13T08:11:35.2916908Z ::add-mask::***
2022-06-13T08:11:35.2918818Z ##[debug]Retrieving Vault Token from v1/auth/jwt/login endpoint
2022-06-13T08:11:35.3811815Z ##[debug]✔ Vault Token successfully retrieved
2022-06-13T08:11:35.3812387Z ::group::Token Info
2022-06-13T08:11:35.3812642Z ##[group]Token Info
2022-06-13T08:11:35.3813410Z ##[debug]Operating under policies: ["default","gha-runner"]
2022-06-13T08:11:35.3814198Z ##[debug]Token Metadata: {"role":"github-sm-image-builder"}
2022-06-13T08:11:35.3814672Z ::endgroup::
2022-06-13T08:11:35.3814921Z ##[endgroup]
2022-06-13T08:11:35.3819620Z ::add-mask::***
2022-06-13T08:11:35.4495051Z ::endgroup::
2022-06-13T08:11:35.4495445Z ##[endgroup]
2022-06-13T08:11:35.4523447Z ##[error]Response code 412 (Precondition Failed)
2022-06-13T08:11:35.4554186Z ##[debug]Node Action run completed with exit code 1
2022-06-13T08:11:35.4568168Z ##[debug]VAULT_TOKEN='***'
2022-06-13T08:11:35.4575999Z ##[debug]Finishing: Get Secrets from vault
```

**Additional context**

The `got` library doesn't consider 412 status code as retryable, see https://github.com/sindresorhus/got/blob/HEAD/documentation/7-retry.md#retry

### PR DESCRIPTION
**Pull request by tomhjp**
Fixes #332 - see issue for details.

412 means the token used has not yet been replicated to the performance replica queried, so we can append it to the list of retryable HTTP codes.

### CODE REVIEW

**Review by swenson — APPROVED** [review]
LGTM

File: src/action.js [hunk 1]
```diff
@@ -26,7 +26,15 @@ async function exportSecrets() {
...
         headers: {},
-        https: {}
+        https: {},
+        retry: {
+            statusCodes: [
+                ...got.defaults.options.retry.statusCodes,
+                // Vault returns 412 when the token in use hasn't yet been replicated
+                // to the performance replica queried. See issue #332.
+                412,
+            ]
```
Comment (calvn) [inline src/action.js:36]: Should we retry on 500's (except 501) too? This is what the Vault Go API client does via [go-retryablehttp](https://github.com/hashicorp/go-retryablehttp#go-retryablehttp).

Comment (tomhjp) [reply src/action.js:36]: I believe 500 and a few other 5XX are already included in the defaults for the `got` dependency: https://github.com/sindresorhus/got/blob/HEAD/documentation/7-retry.md#retry

I'd like to give that a test to be sure we're preserving that default properly though.

Comment (tvoran) [reply src/action.js:36]: How many times will this retry? Should we set a default `limit`? https://github.com/sindresorhus/got/blob/HEAD/documentation/7-retry.md#limit

**Comment by tomhjp** [comment]
I'm going to merge for now, but happy to address any more follow ups in another PR.

**Pull request merged**